File: nodes/python/laser_snapshot.py
#!/usr/bin/env python
import roslib;
roslib.load_manifest('laser_assembler')
import rospy;
from laser_assembler.srv import *
from sensor_msgs.msg import PointCloud

pub = rospy.Publisher("cloud", PointCloud, queue_size=10)
rospy.init_node("laser_snapshot")
rospy.wait_for_service("assemble_scans")
rate = rospy.Rate(.5)
while not rospy.is_shutdown():
	try:
		assemble_scans = rospy.ServiceProxy('assemble_scans', AssembleScans)
		resp = assemble_scans(rospy.Time(0,0), rospy.get_rostime())
		print "Got cloud with %u points" % len(resp.cloud.points)
		pub.publish(resp.cloud)
		rate.sleep()
	except rospy.ServiceException, e:
		print "Service call failed: %s" % e

File: nodes/laser_snapshot.cpp
#include <ros/ros.h>
#include <laser_assembler/AssembleScans2.h>
#include <sensor_msgs/PointCloud2.h>

using namespace laser_assembler;
int main(int argc, char **argv) {
	ros::init(argc, argv, "laser_snapshot");
	ros::NodeHandle n;

	// Wait for laser assembler
	ros::service::waitForService("assemble_scans2");
	
	// Set update frequency (in HZ) to publish point cloud data
	ros::Rate r(1.5);

	// Create publisher
	ros::Publisher pub = n.advertise<sensor_msgs::PointCloud2>("cloud", 2);

	while (ros::ok()) {
		// Server
		AssembleScans2 srv;
		srv.request.begin = ros::Time(0,0);
		srv.request.end = ros::Time::now();
		// Client
		ros::ServiceClient client = n.serviceClient<AssembleScans2>("assemble_scans2");

		// Publish point cloud to topic
		if (client.call(srv)) {
			printf("Got cloud with %u points\n", srv.response.cloud.height*srv.response.cloud.width);
			pub.publish(srv.response.cloud);
		} else {
			printf("Service call failed\n");
		}
		r.sleep();
	}
	return 0;
}